File: test/cpu/CPUMatmulTest.py
import torch

from TestUtils import TestBase


# class CPUMatmul1(TestBase):
#     def test(self):
#         input0 = torch.randn(8, 8)
#         input1 = torch.randn(8, 128)
#         output = torch.matmul(input0, input1)
#         self.test_done(True)


class CPUMatmul1(TestBase):
    def test(self):
        input0 = torch.randn(8, 8)
        input1 = torch.randn(8, 128)
        output = torch.matmul(input0, input1)
        input1 = torch.transpose(input1, 0, 1)
        self.test_done(True)


# class CPUMatmul3(TestBase):
#     def test(self):
#         input0 = torch.randn(8, 8)
#         input1 = torch.randn(8, 128)
#         output = torch.matmul(input0, input1)
#         # input1 = torch.transpose(input1, 0, 1)
#         input0 = torch.transpose(input0, 0, 1)
#
#         self.test_done(True)


if __name__ == '__main__':
    [instance().test() for instance in TestBase.__subclasses__()]

File: test/cpu/CPUMatmulTest.cpp
//
// Created by Xiang Li on 23-10-16.
//
#include "CPUTest.hpp"
#include "backends/cpu/op/CPUMatmul.hpp"
// TEST_F(CPUTest, CPUMatmul1) {
//     SETUP_OP(CPUMatmul, false, false, 4);
//     TENSOR(input0);
//     TENSOR(input1);
//     TENSOR(output);
//     TENSOR(c_output);
//     TEST_LOAD(input0);
//     TEST_LOAD(input1);
//     TEST_LOAD(output);
//     TEST_RESHAPE({input0, input1}, {c_output});
//     TEST_SETUP({input0, input1}, {c_output});
//     PRINT_TENSOR_SHAPES(input0, input1, c_output, output);
//
//     TEST_EXCUTE({input0, input1}, {c_output});
//     //    c_output->printData<float>();
//     COMPARE_TENSOR(c_output.get(), output.get(), true);
// }
TEST_F(CPUTest, CPUMatmul1) {
    SETUP_OP(CPUMatmul, false, true, 4);
    TENSOR(input0);
    TENSOR(input1);
    TENSOR(output);
    TENSOR(c_output);
    TEST_LOAD(input0);
    TEST_LOAD(input1);
    TEST_LOAD(output);
    TEST_RESHAPE({input0, input1}, {c_output});
    TEST_SETUP({input0, input1}, {c_output});
    TEST_EXCUTE({input0, input1}, {c_output});
    PRINT_TENSOR_SHAPES(input0, input1, c_output, output);
    //    c_output->printData<float>();
    COMPARE_TENSOR(c_output.get(), output.get(), true);
}
// TEST_F(CPUTest, CPUMatmul3) {
//     SETUP_OP(CPUMatmul, true, false, 4);
//     TENSOR(input0);
//     TENSOR(input1);
//     TENSOR(output);
//     TENSOR(c_output);
//     TEST_LOAD(input0);
//     TEST_LOAD(input1);
//     TEST_LOAD(output);
//     TEST_RESHAPE({input0, input1}, {c_output});
//     PRINT_TENSOR_SHAPES(input0, input1, c_output, output);
//     TEST_SETUP({input0, input1}, {c_output});
//     TEST_EXCUTE({input0, input1}, {c_output});
////     c_output->printData<float>();
//     COMPARE_TENSOR(c_output.get(), output.get(), true);
// }